### PR TITLE
Us s4 front user

### PR DESCRIPTION
**Pull request by SelinGitt**
All the logic around user (except detailUser)

### CODE REVIEW

**Review by alexboutemy — DISMISSED** [review]

File: src/app/components/articles/article/comment/comment.component.ts [hunk 1]
```diff
@@ -28,32 +33,33 @@ export class ArticleCommentComponent implements OnInit {
...
+        user: { id: loggedInUserString.id }
+      };
+
+      this.commentService.createComment(comment).subscribe(
+        (createdComment) => {
+          this.getCommentsByArticle();
+          form.reset();
+        },
+        (error) => {
+          // Handle the error message returned by the service
```
Comment (Benja-san) [inline src/app/components/articles/article/comment/comment.component.ts:55]: remove dead code

File: src/app/components/user/connection/connection.component.ts [hunk 2]
```diff
@@ -1,29 +1,59 @@
...
+import { confirmPasswordValidator, emailValidator, passwordValidator } from 'src/app/shared/validators/validators';
+import { SnackBarService } from 'src/app/services/snack-bar.service';
 
 @Component({
   selector: 'app-user-connection',
   templateUrl: './connection.component.html',
   styleUrls: ['./connection.component.scss']
 })
-export class ConnectionComponent implements OnInit{
+export class ConnectionComponent implements OnInit {
```
Comment (Benja-san) [inline src/app/components/user/connection/connection.component.ts:14]: optional : I think onInits interface is not required anymore in newest versions of Angular

File: src/app/header/header.component.ts [hunk 3]
```diff
@@ -12,17 +15,27 @@ import { CategoriesApiService } from '../services/categories-api.service';
...
+
+  constructor(public categoriesService: CategoriesApiService, private dialog: MatDialog, private userService: UserService, private snackBarService: SnackBarService) {}
 
   ngOnInit(): void {
     this.categoriesService.getCategories()
       .subscribe(categories => {
         this._categories = categories;
       });
+    if (this.userService.getUser()) {
+      console.log(this.userService.user);
```
Comment (Benja-san) [inline src/app/header/header.component.ts:34]: remove unnecessary logs

File: src/app/shared/validators/validators.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
+import { AbstractControl, ValidationErrors } from '@angular/forms';
+
+export function emailValidator(control: AbstractControl): ValidationErrors | null {
+    const email = control.value;
+
+    // Expression régulière pour la validation de l'adresse e-mail
```
Comment (Benja-san) [inline src/app/shared/validators/validators.ts:6]: remove dead code

File: src/app/shared/validators/validators.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    };
+
+    return !valid ? errors : null;
+}
+
+
+export function passwordValidator(control: AbstractControl): ValidationErrors | null {
+    const password = control.value;
+
+    // Vérification de la longueur minimale
```
Comment (Benja-san) [inline src/app/shared/validators/validators.ts:23]: remove dead code

File: src/app/shared/validators/validators.ts [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    if (password.length < 8) {
+        return {
+            password: {
+                rules: 'Le mot de passe doit contenir au moins 8 caractères',
+            },
+        };
+    }
+
+    // Vérification de la présence d'au moins 1 majuscule, 1 minuscule et 1 chiffre
```
Comment (Benja-san) [inline src/app/shared/validators/validators.ts:32]: remove dead code

File: src/app/shared/validators/validators.ts [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+    if (!digitRegex.test(password)) {
+        return {
+            password: {
+                rules: 'Le mot de passe doit contenir au moins un chiffre',
+            },
+        };
+    }
+
+    // Le mot de passe respecte toutes les règles
```
Comment (Benja-san) [inline src/app/shared/validators/validators.ts:61]: remove dead code

**Review by Benja-san — CHANGES_REQUESTED** [review]

File: src/app/components/user/connection/connection.component.ts (hunk 2 above)
Comment (SelinGitt) [reply src/app/components/user/connection/connection.component.ts:14]: Done.

File: src/app/shared/validators/validators.ts (hunk 4 above)
Comment (SelinGitt) [reply src/app/shared/validators/validators.ts:6]: Done.

File: src/app/shared/validators/validators.ts (hunk 6 above)
Comment (SelinGitt) [reply src/app/shared/validators/validators.ts:32]: Done.

File: src/app/shared/validators/validators.ts (hunk 7 above)
Comment (SelinGitt) [reply src/app/shared/validators/validators.ts:61]: Done.

File: src/app/components/articles/article/comment/comment.component.ts (hunk 1 above)
Comment (SelinGitt) [reply src/app/components/articles/article/comment/comment.component.ts:55]: Done.

File: src/app/header/header.component.ts (hunk 3 above)
Comment (SelinGitt) [reply src/app/header/header.component.ts:34]: Done.

**Review by alexboutemy — APPROVED** [review]

File: src/app/components/user/connection/connection.component.ts [hunk 8]
```diff
@@ -27,10 +27,6 @@ export class ConnectionComponent implements OnInit {
   constructor(private userService: UserService, private usersApiService: UsersApiService, private fb: FormBuilder, private snackBarService: SnackBarService) {
   }
 
-  ngOnInit(): void {
```
[the remark this answers is not in the document]
Comment (SelinGitt) [reply src/app/components/user/connection/connection.component.ts:22]: Done.

**Review by alexboutemy — DISMISSED** [review]

**Review by Benja-san — APPROVED** [review]

**Review by alexboutemy — APPROVED** [review]

**Pull request merged**